auth/auth.go

```diff
@@ -163,3 +171,7 @@ func makeAuthType(s string) (AuthType, error) {
 	}
 	return AuthType(""), ErrMalformedAuthorization
 }
+
+func bytesToString(b []byte) string {
+	return *(*string)(unsafe.Pointer(&b))
```

Comment:
does this add a ref to b? (so that it's not nilref'd when b is GC'd)

Reply:
It's just literally a type cast of the slice pointer to a string pointer, then dereferencing the string pointer. But this common trick is used everywhere and in go1.20 they made a bit more formal way.

I similarly have this generic: https://github.com/planetscale/edge-gateway/blob/main/common/mysqlserver/mysqlserver_unsafe.go#L27-L29

In go1.20 there's a newer `unsafe.String` https://pkg.go.dev/unsafe#String

https://cs.opensource.google/go/go/+/refs/tags/go1.20:src/strings/builder.go;l=46-49
vs
https://cs.opensource.google/go/go/+/refs/tags/go1.19.5:src/strings/builder.go;l=46-49

I don't want to require go1.20 for this, so just using the older style trick.

Vitess also has this in a few places: https://github.com/vitessio/vitess/blob/main/go/hack/hack.go#L34

Reply:
The only real risk with doing this is when the underlying bytes may change or something from underneath it since it's just casting the poiinter and they share the same memory. So it just needs to be done over trusted memory.

In this case, the byte slice is allocated explicitly already, and we don't need another copy into a string. The copy is generally safer if the input and output are unknown.